### PR TITLE
zrythm: init at 1.0.0-alpha.26.0.13

### LINKED ISSUE #98871: ZRythm
**Project description**

A highly automated and intuitive digital audio workstation.

**Metadata**

* homepage URL: https://www.zrythm.org
* source URL: https://github.com/zrythm/zrythm
* license: AGPLv3+
* platforms: linux

### PR DESCRIPTION
**Pull request by tshaynik**
Add a package for the zrythm Digital Audio Workstation.

Closes #98871

Built on work done by @andreasfelix on #129457 but adds additional
missing dependencies, uses the latest zrythm
version, makes use of dependencies that have since been added to
nixpkgs, and resolves the issue of the crash on start that previous
packaging attempt was encountering.

###### Motivation for this change

Add a package for the zrythm Digital Audio Workstation.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by magnetophon — APPROVED** [review]
LGTM!
Builds and runs here.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -11639,6 +11639,12 @@
     email = "4ZKTUB6TEP74PYJOPWIR013S2AV29YUBW5F9ZH2F4D5UMJUJ6S@hash.domains";
     name = "Thomas Strobel";
   };
+  tshaynik = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:11642]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/applications/audio/zrythm/default.nix [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+, zstd
+}:
+
+let
+  pname = "zrythm";
+  version = "1.0.0-alpha.26.0.13";
+in
+stdenv.mkDerivation {
+  inherit pname;
+  inherit version;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/zrythm/default.nix:69]:
```suggestion
stdenv.mkDerivation rec {
  pname = "zrythm";
  version = "1.0.0-alpha.26.0.13";```

File: pkgs/applications/audio/zrythm/default.nix [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+  nativeBuildInputs = [
+        git
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/zrythm/default.nix:72]: Why do we need git? We probably need to patch the usage.

File: pkgs/applications/audio/zrythm/default.nix [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        serd
+        sord
+        sratom
+        vamp-plugin-sdk
+        xdg-utils
+        xxHash
+        zstd
+      ];
+
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/zrythm/default.nix:134]: src goes before any buildInputs.

File: pkgs/applications/audio/zrythm/default.nix [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix GSETTINGS_SCHEMA_DIR : "$out/share/gsettings-schemas/${pname}-${version}/glib-2.0/schemas/"
+    )
+  '';
+
+  meta = with lib; {
+    homepage = "https://www.zrythm.org";
+    description = "highly automated and intuitive digital audio workstation";
+    maintainers = [ maintainers.tshaynik maintainers.magnetophon ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/zrythm/default.nix:171]:
```suggestion
    maintainers = with maintainers; [ tshaynik magnetophon ];
```

File: pkgs/applications/audio/zrythm/default.nix [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix GSETTINGS_SCHEMA_DIR : "$out/share/gsettings-schemas/${pname}-${version}/glib-2.0/schemas/"
+    )
+  '';
+
+  meta = with lib; {
+    homepage = "https://www.zrythm.org";
+    description = "highly automated and intuitive digital audio workstation";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/zrythm/default.nix:170]:
```suggestion
    description = "Highly automated and intuitive digital audio workstation";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please format the package file with nixpkgs-fmt.

File: pkgs/applications/audio/zrythm/default.nix (hunk 2 above)
Comment (tshaynik) [reply pkgs/applications/audio/zrythm/default.nix:69]: Is there a reason that this is preferred? It seems like they're equivalent, and I've seen `rec` referred to as an anti-pattern (i.e. https://nix.dev/anti-patterns/language#rec-expression)
I'm happy to change it though.

File: pkgs/applications/audio/zrythm/default.nix (hunk 4 above)
Comment (tshaynik) [reply pkgs/applications/audio/zrythm/default.nix:66]: fixed!

File: pkgs/applications/audio/zrythm/default.nix (hunk 3 above)
Comment (tshaynik) [reply pkgs/applications/audio/zrythm/default.nix]: Removed git, and it didn't break. Thanks!

**Comment by tshaynik** [comment]
> Please format the package file with nixpkgs-fmt.

I formatted with nixpkgs-fmt, addressed all of your points, and rebased. Let me know what you think @SuperSandro2000.
